TeamShiksha/logoexecutive
# feat/UpdateProfileLatest

### Summary

Added functionality to allow users to change their name and email accounts.
Addressed Issue #45 

### Description

#### POST API : router.post("/",authMiddleware, updateProfileController);
1. Receives the current email ID and new data (name, email) in the request body.
2. Make the changes in the first name and last name if required and sends a verification email to change the email.

### How to Test the Changes

#### Testing Steps:
1. Run the application locally and ensure it's functioning properly.
2. Use an API testing tool (e.g., Postman) to trigger the following API endpoints:
3. Send a POST request to /update-profile with required payload and query params.
4. Check the email inbox for the received email and click the verification link.
5. Verify that the user details are updated by sending a GET request to router.get("/verify", verifyTokenController);

#### Test Configuration:

1. Ensure the test environment has access to the SMTP server specified in the HOST variable.
2. Validate that the .env file is correctly configured and loaded by the application to use the provided email credentials for sending emails.

### Context (Optional)

This feature addresses the need for users to update their name and email details securely within the application.

### Screenshots or Recordings (Optional)

![SS1 Update Profile](https://github.com/FrontendArmy/logoexecutive-backend/assets/99191853/1d8f75d7-da93-4aa8-bec8-df2cc06f4637)

![SS2 Update Profile](https://github.com/FrontendArmy/logoexecutive-backend/assets/99191853/057c1251-89d2-4d2c-a430-8938c237f731)

![SS3 Update Profile](https://github.com/FrontendArmy/logoexecutive-backend/assets/99191853/e2bf3411-9caa-4a36-8b62-9ea8a64f60ad)



## Checklist
